sql_injection.js

```diff
@@ -1,8 +1,9 @@
 function checkLogin(req, db) {
+  const password_token = 'super-secret-f8ed84e8f41e4146403dd4a6bbcea5e418d24i7';
   const sqlQuery =
     "SELECT email FROM credentials WHERE " +
-    "email = ? AND " +
-    "password = ?";
+    "email = '" + req.body.email + "' AND " +
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg" width="20"/></sub>&nbsp; SQL Injection

Unsanitized input from the HTTP request body flows into query, where it is used in an SQL query. This may result in an SQL Injection vulnerability.

<h6>
Line 5 | <a href="https://cwe.mitre.org/data/definitions/798.html">CWE-798</a> | <a href="https://cwe.mitre.org/data/definitions/259.html">CWE-259</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 562</a></h6>

<details>

<summary><b>Data flow: 8 steps</b></summary>


**`Step 1 - 3`**

https://github.com/trailblazing-dx/tiago_app.dev.snyk_github_app_nodejs-goof/blob/7365bfe21704b66bc15456f83df582bb03210d38/sql_injection.js#L5

`Step 4` [sql_injection.js#L3](https://github.com/trailblazing-dx/tiago_app.dev.snyk_github_app_nodejs-goof/blob/7365bfe21704b66bc15456f83df582bb03210d38/sql_injection.js#L3)

`Step 5`  [sql_injection.js#L8](https://github.com/trailblazing-dx/tiago_app.dev.snyk_github_app_nodejs-goof/blob/7365bfe21704b66bc15456f83df582bb03210d38/sql_injection.js#L8)

`Step 6`  [sql_injection.js#L11](https://github.com/trailblazing-dx/tiago_app.dev.snyk_github_app_nodejs-goof/blob/7365bfe21704b66bc15456f83df582bb03210d38/sql_injection.js#1)

`Step 7-8` https://github.com/trailblazing-dx/tiago_app.dev.snyk_github_app_nodejs-goof/blob/7365bfe21704b66bc15456f83df582bb03210d38/sql_injection.js#L12

</details>